hydromt/models/components/mesh.py

```diff
@@ -199,113 +554,41 @@ def setup_mesh2d_from_raster_reclass(
-        # Checks on data
-        if not isinstance(data, (xu.UgridDataArray, xu.UgridDataset)):
-            raise ValueError(
-                "New mesh data in set_mesh should be of type xu.UgridDataArray"
-                " or xu.UgridDataset"
-            )
-        if isinstance(data, xu.UgridDataArray):
-            if name is not None:
-                data = data.rename(name)
-            elif data.name is None:
-                raise ValueError(
-                    f"Cannot set mesh from {str(type(data).__name__)} without a name."
-                )
-            data = data.to_dataset()
-
-        # Checks on grid topology
-        # TODO: check if we support setting multiple grids at once. For now just one
-        if len(data.ugrid.grids) > 1:
-            raise ValueError(
-                "set_mesh methods only supports adding data to one grid at a time."
-            )
-        if grid_name is None:
-            grid_name = data.ugrid.grid.name
-        elif grid_name != data.ugrid.grid.name:
-            data = workflows.rename_mesh(data, name=grid_name)
+            mesh2d=self.mesh_grids[grid_name],
+            reclass_variables=reclass_variables,
+            fill_method=fill_method,
+            resampling_method=resampling_method,
+            rename=rename,
+            logger=self.logger,
+        )
 
-        # Check if new grid_name
-        if grid_name not in self.mesh_names:
-            new_grid = True
-        else:
-            new_grid = False
+        self.set(uds_sample, grid_name=grid_name, overwrite_grid=False)
+
+        return list(uds_sample.data_vars.keys())
 
-        # Adding to mesh
-        if self.mesh is None:  # NOTE: mesh is initialized with None
+    def _add_mesh(
+        self, data: xu.UgridDataset, grid_name: str, overwrite_grid: bool
+    ) -> Optional[CRS]:
+        if self._data is None:  # NOTE: mesh is initialized with None
             # Check on crs
             if not data.ugrid.grid.crs:
                 raise ValueError("Data should have CRS.")
-            self._mesh = data
+            self._data = data
+            return None
         else:
             # Check on crs
             if not data.ugrid.grid.crs == self.crs:
-                raise ValueError("Data and self.mesh should have the same CRS.")
+                raise ValueError("Data and self.data should have the same CRS.")
             # Save crs as it will be lost when converting to xarray
             crs = self.crs
             # Check on new grid topology
             if grid_name in self.mesh_names:
                 # check if the two grids are the same
-                if (
-                    not self.mesh_grids[grid_name]
-                    .to_dataset()
-                    .equals(data.ugrid.grid.to_dataset())
-                ):
-                    if not overwrite_grid:
+                if not self._grid_is_equal(grid_name, data):
+                    if not overwrite_grid:  # TODO: add is_overwrite_mode
```

Comment:
Fix todo, or make issue?

Reply:
Why was the is_overwrite_mode added? Calling the whole ``set`` method means you are overwriting anyway if not the grid then the data...

Reply:
I removed is_overwrite_mode. Not sure why I added that